pallets/api/src/fungibles/mod.rs

```diff
@@ -318,7 +327,15 @@ pub mod pallet {
 			admin: AccountIdOf<T>,
 			min_balance: BalanceOf<T>,
 		) -> DispatchResult {
-			AssetsOf::<T>::create(origin, id.into(), T::Lookup::unlookup(admin), min_balance)
+			AssetsOf::<T>::create(
+				origin.clone(),
+				id.clone().into(),
+				T::Lookup::unlookup(admin.clone()),
+				min_balance,
+			)?;
+			let creator = ensure_signed(origin)?;
```

Comment:
I'd stick this as the first statement for consistency.

Reply:
see last commit